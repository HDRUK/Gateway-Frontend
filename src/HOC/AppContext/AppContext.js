import React, { Component } from "react";
import PropTypes from "prop-types";
import hdruk_logo_black from "../../assets/hdruk_black.png";

export const AppContext = React.createContext();

export const AppContextConsumer = AppContext.Consumer;

class AppContextProvider extends Component {
    newsItems = {
        newsItemOne: {
            image: "",
            description: "HDR release news of new partners in exciting new digital project",
            readMore: "http://localhost:3000"
        },
        newsItemTwo: {
            image: "",
            description: "HDR release news of new partners in exciting new digital project",
            readMore: "http://localhost:3000"
        },
        newsItemThree: {
            image: "",
            description: "HDR release news of new partners in exciting new digital project",
            readMore: "http://localhost:3000"
        }
    };

    images = {
        logoHDR: hdruk_logo_black
    };

    filter = [];
    textItems = { searchHeader: "What health data do you need?" };

    searchPageStates = {
        form: "form",
        results: "results"
    };

    itemRef = React.createRef();

    state = {
        counter: 0,
        searchPageState: this.searchPageStates.form,
        modalVisibility: false,
        filterLocation: 0,
        filterId: 0
    };

    returnSearchResults = event => {
        if (event.key === "Enter") {
            this.setState({
                searchPageState: this.searchPageStates["results"]
            });
        }
    };

    outsideRange = (number, target, range) => {
        return number < target - range || number > target + range;
    };

    setFilterLocation = () => {
        this.itemRef.current &&
            this.outsideRange(this.itemRef.current.getBoundingClientRect().y, this.state.filterLocation, 11) &&
            this.setState({ filterLocation: this.itemRef.current.getBoundingClientRect().y });
    };

    setFilterId = props => {
        this.setState({ filterId: props });
    };

    counterFunc = () => {
        this.setState({
            counter: this.state.counter + 1
        });
    };

    addFilter = props => {
        this.filters.append(props);
    };

    toggleModal = () => {
        this.setState(
            {
                modalVisibility: !this.state.modalVisibility
            },
            () =>
                this.state.modalVisibility
                    ? document
                          .getElementById("main-side-nav")
                          .childNodes[1].addEventListener("scroll", this.setFilterLocation)
                    : document
                          .getElementById("main-side-nav")
                          .childNodes[1].removeEventListener("scroll", this.setFilterLocation)
        );
    };

    render() {
        return (
            <AppContext.Provider
                value={{
                    state: this.state,
                    counterFunc: this.counterFunc,
                    newsItems: this.newsItems,
                    images: this.images,
                    textItems: this.textItems,
                    returnSearchResults: this.returnSearchResults,
                    setFilterLocation: this.setFilterLocation,
                    toggleModal: this.toggleModal,
                    setFilterId: this.setFilterId,
                    itemRef: this.itemRef
                }}
            >
                {this.props.children}
            </AppContext.Provider>
        );
    }
}

AppContextProvider.propTypes = {
    children: PropTypes.node
};

export default AppContextProvider;
