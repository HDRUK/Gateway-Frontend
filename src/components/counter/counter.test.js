import React from "react";
import { create, act } from "react-test-renderer";
import Counter from "./counter";

import { AppContext } from "../../HOC/AppContext/AppContext.js";
import context from "../../__mocks__/AppContextMock.js";
import mocks from "../../__mocks__/ApolloMock.js";

import { MockedProvider } from "@apollo/react-testing";

const wait = ms => new Promise(resolve => setTimeout(resolve, ms));

describe("<LandingPage> ", () => {
    let renderedComponent;

    describe("load query", () => {
        beforeEach(() => {
            act(() => {
                renderedComponent = create(
                    <MockedProvider mocks={mocks}>
                        <AppContext.Provider value={context}>
                            <Counter />
                        </AppContext.Provider>
                    </MockedProvider>
                );
            });
        });

        it("should test the counter loading state snapshot", () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it("should render the query result", async () => {
            await act(async () => await wait(1000));
            const p = renderedComponent.root.findAllByType("p");

            expect(p[0].props.children).toEqual(0);
            expect(p[1].props.children).toEqual(["Jurassic Park", ": ", "Michael Crichton"]);
        });
    });

    describe("when counter clicked", () => {
        beforeEach(async () => {
            await act(async () => await wait(1000));
            const p = renderedComponent.root.findAllByType("p");
            await act(async () => p[0].props.onClick());
        });

        it("should call counterFunc", async () => {
            expect(context.counterFunc).toHaveBeenCalled();
        });
    });

    describe("load failing query", () => {
        beforeEach(async () => {
            let errorMocks = mocks;
            errorMocks[0].result = null;
            errorMocks[0].error = new Error("Failed to load query");
            act(() => {
                renderedComponent = create(
                    <MockedProvider mocks={mocks}>
                        <AppContext.Provider value={context}>
                            <Counter />
                        </AppContext.Provider>
                    </MockedProvider>
                );
            });
            await act(async () => await wait(1000));
        });

        it("should display error", () => {
            const div = renderedComponent.root.findByType("div");
            expect(div.props.children).toEqual("Error :(");
        });
    });
});
