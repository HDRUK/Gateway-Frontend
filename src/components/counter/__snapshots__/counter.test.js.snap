// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage>  load query should test the counter loading state snapshot 1`] = `
<p>
  Loading ...
</p>
`;
